### PR TITLE
feat(psl): GA `fullTextIndex` for every supported provider: `mysql`, `mongodb`

### PR DESCRIPTION
**Pull request by jkomyno**
This PR:
- Contributes to https://github.com/prisma/team-orm/issues/1327
- Deprecates https://github.com/prisma/prisma-engines/pull/5031
- GAs `fullTextIndex`
- Adapts existing tests
- Using `previewFeatures = ["fullTextIndex"]` in `schema.prisma` now results in a validation warning:
  ```
  Preview feature "fullTextIndex" is deprecated. The functionality can be used without specifying it as a preview feature.
  ```

---

This PR implies that introspection of existing `@fullText` indexes works.
For example, with MongoDB:
- Let `$TEST_MONGO_URI` be `"mongodb://localhost:27018/tests"`
- Let's first run `prisma db push` on the following schema, which we call `./prisma/schema.prisma`:

  ```prisma
  // ./prisma/schema.prisma
  generator client {
    provider = "prisma-client-js"
  }
  
  datasource db {
    provider = "mongodb"
    url      = "mongodb://localhost:27018/tests"
  }
  
  model User {
    id   String @id @map("_id") @db.ObjectId
    name String
  
    @@fulltext([name])
  }
  ```
- Let's delete the `model User`:
  ```diff
  // ./prisma/schema.prisma
  generator client {
    provider = "prisma-client-js"
  }
  
  datasource db {
    provider = "mongodb"
    url      = "mongodb://localhost:27018/tests"
  }

  - model User {
  -   id   String @id @map("_id") @db.ObjectId
  -   name String
  -
  -    @@fulltext([name])
  -  }
  ```
- Now, we introspect the existing database. Let's run `prisma db pull --url $TEST_MONGO_URI`, which yields the following result:

  ```prisma
  // ./prisma/schema.prisma
  generator client {
    provider = "prisma-client-js"
  }
  
  datasource db {
    provider = "mongodb"
    url      = "mongodb://localhost:27018/tests"
  }
  
  model User {
    id   String @id @map("_id") @db.ObjectId
    name String
  
    @@fulltext([name])
  }
  ```

As you can see, the full-text index on the `name` field is preserved as `@@fulltext([name])`

### CODE REVIEW

File: schema-engine/connectors/mongodb-schema-connector/src/client_wrapper.rs [hunk 1]
```diff
@@ -9,11 +9,10 @@ use schema_connector::{ConnectorError, ConnectorResult};
 pub struct Client {
     inner: mongodb::Client,
     db_name: String,
-    preview_features: BitFlags<PreviewFeature>,
 }
 
 impl Client {
-    pub async fn connect(connection_str: &str, preview_features: BitFlags<PreviewFeature>) -> ConnectorResult<Client> {
+    pub async fn connect(connection_str: &str, _preview_features: BitFlags<PreviewFeature>) -> ConnectorResult<Client> {
```
Comment (aqrln) [inline schema-engine/connectors/mongodb-schema-connector/src/client_wrapper.rs:15]: Should we remove the parameter as well?

**Review by aqrln — COMMENTED** [review]
Looks great, one comment/question and a few missing snapshot updates that are currently failing on CI, and we're good to merge.

File: schema-engine/connectors/mongodb-schema-connector/src/client_wrapper.rs (hunk 1 above)
Comment (jkomyno) [reply schema-engine/connectors/mongodb-schema-connector/src/client_wrapper.rs:15]: I would recommend to leave it here. Both the `mongodb-schema-connector` and the `sql-schema-connector` rely on preview features, and we'll likely need this again in the future. Removing this `preview_features` parameter (and its propagated calls) to add it again in the future just adds noise, imho.

**Comment by jkomyno** [comment]
> Looks great, one comment/question and a few missing snapshot updates that are currently failing on CI, and we're good to merge.

All snapshots pass since https://github.com/prisma/prisma-engines/pull/5052/commits/f413ba4b15f3d3a850d9325d2bed84f5fc25dbfd

**Review by aqrln — APPROVED** [review]
nice!

**Pull request merged**